tests/models/gemma2/test_modeling_gemma2.py

```diff
@@ -161,3 +164,29 @@ def test_model_9b_pipeline_bf16(self):
 
         self.assertEqual(output[0][0]["generated_text"], EXPECTED_TEXTS[0])
         self.assertEqual(output[1][0]["generated_text"], EXPECTED_TEXTS[1])
+
+    @require_read_token
+    @require_flash_attn
+    @require_torch_gpu
+    @mark.flash_attn_test
+    @slow
+    def test_model_9b_flash_attn(self):
+        # See https://github.com/huggingface/transformers/issues/31953 --- flash attn was generating garbage for gemma2, especially in long context
+        # NOTE: the quality is a lot better whan fp16 is used, and worse for bf16
```

Comment:
Could you elaborate more on this? Specifically, does using `flash_attention_2` negatively impact the quality of the model's output with `bfloat16` compared to using `eager` mode?

Reply:
Oh, this was true until I discovered a bug I made. It doesn't matter much which precision is used, removed the comment